SETUP/ci/check_security.php

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env php
+<?php
+// Check all .php and .inc files for security best-practices.
+
+$relPath = "../../pinc/";
+include_once($relPath."misc.inc");
+
+$basedir = $argv[1] ?? "../../";
+
+// List of files that can contain system/exec calls
+// TODO: Likely few or none of these should contain the calls either --
+//       opting for Symfony Process() instead -- but this ensures that no
+//       others are added from the current set until they get updated.
+$ok_system_calls = [
+    "crontab/CleanUploadsTrash.inc",
+    "locale/translators/index.php",
+    "pinc/archiving.inc",
+    "pinc/forum_interface_phpbb3.inc",
+    "pinc/languages.inc",
+    "pinc/phpbb3.inc",
+    "pinc/POFile.inc",
+    "pinc/Project.inc",
+    "pinc/project_quick_check.inc",
+    "pinc/upload_file.inc",
+    "tools/project_manager/add_files.php",
+    "tools/project_manager/show_project_stealth_scannos.php",
+    "tools/proofers/images_index.php",
+];
+
+// List of files that can contain mysqli_error() calls
+$ok_mysqli_error_calls = [
+    "pinc/DPDatabase.inc",
+];
+
+$basedir .= endswith($basedir, "/") ? "" : "/";
+$files = get_all_php_files($basedir);
+foreach ($files as $file) {
+    // If it's in the SETUP directory, skip it
+    if (startswith($file, "SETUP/")) {
```

Comment:
would it be simpler to have an array of files startswith to skip?

Reply:
Yeah, and it's not as though the comments are adding anything to those if statements either. Done.